### PR TITLE
add a helper fn and necessary dependencies to query gemini for price …

### PR DESCRIPTION
**Pull request by zancas**
…information

This is the first commit towards removing price-acquisition from our `lightwalletd` interface.

### CODE REVIEW

**Comment by dannasessha** [comment]
ACK, I'll look to review this soon.

**Review by dannasessha — COMMENTED** [review]
Notes attached.

File: lib/src/lightclient.rs [hunk 1]
```diff
@@ -650,28 +653,25 @@ impl LightClient {
     pub fn get_server_uri(&self) -> http::Uri {
         self.config.server.clone()
     }
-
+    async fn get_and_calc_price_from_gemini(&self) -> Decimal {
+        let trades = reqwest::get("https://api.gemini.com/v1/trades/zecusd")
```
[the remark this answers is not in the document]
Comment (zancas) [reply lib/src/lightclient.rs:657]: `gemini` has wider visibility than Aditya's version of lightwalletd.   It's a much safer source of truth.

File: lib/src/lightclient.rs [hunk 2]
```diff
@@ -650,28 +653,25 @@ impl LightClient {
     pub fn get_server_uri(&self) -> http::Uri {
         self.config.server.clone()
     }
-
+    async fn get_and_calc_price_from_gemini(&self) -> Decimal {
```
[the remark this answers is not in the document]
Comment (zancas) [reply lib/src/lightclient.rs:656]: Nowhere, and in fact this doesn't need to be a method.

**Comment by zancas** [comment]
Yes..  this should be marked Draft.

**Comment by zancas** [comment]
Work to do:

* median instead of mean
* determine where (if anywhere to store price)
* check into price skew bug

File: lib/src/lightclient.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dannasessha) [reply lib/src/lightclient.rs]: Yup, okay, I agree. Should we make a comment on the readme, and/or in the help for the command?

**Comment by zancas** [comment]
@dannasessha can you sanity check the regtest `zingo-cli` against this version of the code?   Do the price-related commands work?  How do they behave?

**Review by AloeareV — CHANGES_REQUESTED** [review]
This looks good, other than the one change I've requested

File: lib/src/lightclient.rs [hunk 3]
```diff
@@ -1198,86 +1226,14 @@ impl LightClient {
 
     async fn update_current_price(&self) {
```
[the remark this answers is not in the document]
Comment (AloeareV) [reply lib/src/lightclient.rs:1227]: I don't think unconditionally, but there should be an option for it. If the price is ten minutes out of date, it likely makes sense to not fetch. If the price is ten weeks out of date, it should definitely be fetched. I think the user should have control of this.

[the remark this answers is not in the document]
Comment (zancas) [reply lib/src/lightclient.rs:1227]: What if we just have 1 interface.   It works unconditionally, but only when the user pushes the button?

[the remark this answers is not in the document]
Comment (AloeareV) [reply lib/src/lightclient.rs:1227]: Sure. Although, we want the user to be able to check the price they have on file without making another request to the server.

File: lib/src/lightclient.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dannasessha) [reply lib/src/lightclient.rs]: followed up on this at https://github.com/zancas/zingolib/pull/1

File: cli/src/main.rs [hunk 4]
```diff
@@ -55,8 +54,245 @@ pub fn main() {
 
     let regtest_mode_enabled = matches.is_present("regtest");
     if regtest_mode_enabled {
-        regtest::launch()
-        // do the regtest
+        use std::ffi::OsString;
```
Comment (AloeareV) [inline cli/src/main.rs:57]: Is this a git rebase reversion?

Comment (zancas) [reply cli/src/main.rs]: You caught me...  but it's already fixed.

**Review by AloeareV — APPROVED** [review]
Tested locally (with @dannasessha's new regtest mode, to boot!) and it looks great! Merging!

**Pull request merged**